sound/soc/sof/intel/hda-dsp.c

```diff
@@ -417,9 +425,29 @@ static int hda_dsp_set_D0_state(struct snd_sof_dev *sdev,
 		    !hda_enable_trace_D0I3_S0 ||
 		    sdev->system_suspend_target != SOF_SUSPEND_NONE)
 			flags = HDA_PM_NO_DMA_TRACE;
+
+		/* disable DMAs if needed */
+		if (hda_S0ix_host_dma_stop || chip->S0ix_host_dma_stop_required) {
+			flags |= HDA_PM_HOST_DMA_STOP_NOTIFICATION;
+
+			/* the DMAs are stopped after the PM_GATE IPC */
+		}
 	} else {
 		/* prevent power gating in D0I0 */
 		flags = HDA_PM_PPG;
+
+		/* restart DMAs if needed */
+		if (hda_S0ix_host_dma_stop || chip->S0ix_host_dma_stop_required) {
+			flags |= HDA_PM_HOST_DMA_RESTART_NOTIFICATION;
+
+			ret = snd_sof_stream_trigger_suspend_ignored(sdev, SNDRV_PCM_TRIGGER_START);
+
+			/* on errors we still exit S0ix and try to restart the DSP */
```

Comment:
how will restarting the DSP be triggered? This isn't something, added by this PR, right?

Reply:
the wording is inaccurate, it should be restore the full DSP state. that's handled by the PM_GATE IPC.